CRM/Contribute/BAO/Query.php

```diff
@@ -108,12 +97,6 @@ public static function select(&$query) {
       $query->_tables['contribution_batch'] = 1;
     }
 
-    if (!empty($query->_returnProperties['contribution_source'])) {
```

Comment:
This can be handled in the main loop based on metadata due to the removal of skipping that loop for 'contribution_' (above)